README
Mac-On-Linux 0.9.73-SVN - http://sourceforge.net/p/mac-on-linux/code/HEAD/tree/

Mac-on-Linux is a Linux/PPC program that virtualizes MacOS or MacOSX up to 10.3.x in Linux.

----------------

cd mol-src
make menuconfig 
make 
make install

Read the documentation!

---------------
Changes: 

08/09/2015: 
	* Make MOL recognize newewer kernels.


mol-src/CHANGELOG
MOL Changelog

2007-12-25 Joseph Jezak
	* Removed the ethertap driver
2007-12-24 Joseph Jezak
	* Build fixes for newer kernels
	* Moved the tmp directory for tmp_offsets building
2007-07-29 Joseph Jezak
	* Revert FC7 header fix, it's breaking on other distros
2007-07-23 Joseph Sacco
	* Mark 7448 as able to use Altivec
2007-07-23 Joseph Jezak
	* Added PVRs for some more CPUs
	* Stop the build on x86 to prevent bug reports
	* Added some build stuff for ppc64 (don't get excited...)

mol-0.9.72.1
2007-06-25 Robert Story
	* Fix building on FC7 by adding missing headers
2007-06-21 Andreas Schwab
	* Additional fixes for find_physical_rom
2007-05-22 Joseph Jezak
	* Fix bus frequency reporting
2007-05-15 Joseph Jezak
	* 2.6.22 kmod fixes
2007-04-09 Joseph Sacco
	* Remove TUN from defconfig (again)
2007-04-07 Risto Suominen
	* Fix XDGA off by one error
2007-03-12 Joseph Jezak 
	* Update netdriver Kconfig
	* Re-arrange the Kconfig menus
	* Add SDL master driver
	* Add SDL sound driver
	* Remove configure directory
2007-03-06 Joseph Jezak 
	* OSX Compatibility fixes

mol-0.9.72
2007-03-02 Joseph Jezak 
	* Release
2007-02-09 Joseph Jezak 
	* Fixed tilde key
2007-02-05 Joseph Jezak 
	* Fix CPU clock speed reporting
2007-02-05 Joseph E. Sacco
	* Add missing lock and log dirs to RPM generation
2007-02-02 Joseph E. Sacco
	* Fix building RPMS
	* Add Sheep and TUN back to defconfig
2007-01-31 didier
	* VNC - Redraw request fix
2007-01-26 Gerhard Pircher
	* Remove cache coherency requirement for AmigaOne support
2007-01-17 Joseph Jezak 
	* usleep -> nanosleep
2007-01-16 Joseph Jezak 
	* Portability fixes

mol-0.9.72_pre2
2007-01-15 Joseph Jezak 
	* Added nvram.x installation back
	* Reverted BootX init.S changes (fixes cdboot bug)
	* Fix bootstrap check
2007-01-14 Joseph Jezak 
	* Fix dmdma.c strict aliasing warning
	* Make sure that log and lock dirs are created
2007-01-13 Joseph Jezak 
	* Integrated BootX with the build system
	* Added check for zlib
2007-01-12 Joseph Jezak 
	* Fixed test invocations (== is invalid)
	* Fixed VT Switching
2007-01-10 Joseph Jezak 
	* Updated BootX's init.S to match MOL's version
2007-01-08 Joseph Jezak 
	* Removed Sheep from defconfig
	* Fixed Ubuntu -fno-stack-protector bug
	* Removed nvram.x, fix doesn't really work
	* Some docs cleanups, changed the website address
	* Removed libimport script
	* Fixed another MacOS segfault on startup
	* Ensured that all read/write/seek disk pointers are set properly
2007-01-06 Joseph Jezak 
	* Removed some debugging text
	* Fixed MacOS ethernet softlock (Thanks SolraBizna for noticing)
	* Fixed MacOS segfault on startup (Thanks to SolraBizna for noticing)
2006-12-29 Joseph Jezak 
	* Changed BootX Read to pass addresses instead of casting them as long
	* Fixed BootX read size bug preventing booting on QCOW/DMG
	* Made BootX compile look like MOL compile
2006-11-24 Joseph Jezak 
	* Cleaned up a bunch of gcc4 issues
	* Removed TUN from defconfig
	* Updated network driver help info
	* Integrated patch from Marcus Comstedt for USB mice
	* Removed linux/compiler.h from USB to fix Ubuntu issue
2006-11-10 Joseph Jezak 
	* Fix mol-img bug
	* Added support to recognize HFSX disks
2006-11-08 Joseph Jezak 
	* Added bootx source to main MOL build (with backports from newer BootX)
	* Fixed gcc4 include issue

mol-0.9.72_pre1
2006-11-06 Joseph Jezak 
	* Fix build as user tmp-offsets issue
	* Added mol-img to default build
	* Added mol-img for creating qcow and raw images
	* Fixed qcow issues
2006-10-28 Joseph Jezak 
	* Added block driver abstraction layer
	* Added qcow disk support
	* Added compressed dmg support

mol-0.9.71.1
2006-10-28 Joseph Jezak 
	* Added new molvconfig manpage from debian
	* Compile fixes for 2.6.19

mol-0.9.71
2006-10-02 Joseph Jezak 
	* Fixed ablk bug introduced by gcc4 fixes

mol-0.9.71_pre9
2006-09-05 Joseph Jezak 
	* Lots of GCC 4.x warning fixes
	* Compile fixes for kernels 2.6.17 and newer

mol-0.9.71_pre8
2006-01-26 Joseph Jezak 
	* Fixed mol kmod naming issue, same as last version's netdev fix
	* Added video resolution settings for new PBs and Apple Cinema Displays 

mol-0.9.71_pre7
2006-01-10 Joseph Jezak 
	* Reverted the "openbios" back to the "of" binary, fixes booting OS9
	* Fixed netdev install directory with 2.6 kernels
	* Fixed version variable

mol-0.9.71_pre6
2005-12-01 Joseph Jezak 
	* Removed the 2.6.15 fix, it's fixed upstream now
	* Included typo fix in src/netdriver/Makefile.26 from Mattias Nissler

mol-0.9.71_pre5
2005-11-27 Joseph Jezak 
	* Updated mods.mkext with OSX 10.2 compatible version
	* Fixed building with 2.6.15
	* Included some more debian patches
	* Included build fixes for 2.4 from Hans-Martin Mosner

mol-0.9.71_pre4
2005-10-12 Joseph Jezak  
	* Big filesystem fix is applied in build system instead of the ebuild
	* Tiger nvram fix is applied in the build system instead of the ebuild 
	* Fix for improper binutils handling of some ASM (thanks Bin Zhang)
	* Fixed versions and a few minor typos

mol-0.9.71_pre3
2005-10-12 Joseph Jezak  
	* Tiger Acceleration fix by adding NW compatible NVRAM image
	* GCC4 compile fixes

mol-0.9.71_pre2
2005-10-12 Joseph Jezak  
	* Based on MOL BK development pull
	* Build fix for 2.6.9
	* Build fix for 2.6.10 
	* Build fix for 2.6.12
	* Scheduler Fixes (fixes lag in newer kernels)
	* .tmp_offset patch fixes sandbox issues on Gentoo
	* Tiger support (Mattias Nissler's patch)
	* PCI Proxy support
	* MOL module loading name fix
	* Tiger 8GB Disk install patch
	* 96MB BootX fix
	* XSHM fix
	* Sheep building typo fixed
	* Probably other minor things


mol-kernel-merge/rvec.h
/* 
 *   Creation Date: <2001/01/26 21:33:45 samuel>
 *   Time-stamp: <2004/02/08 20:08:20 samuel>
 *   
 *	<return_vectors.h>
 *	
 *	Possible mac-return vectors (see mainloop.S)
 *   
 *   Copyright (C) 2000, 2001, 2002, 2003, 2004 Samuel Rydh (samuel@ibrium.se)
 *   
 *   This program is free software; you can redistribute it and/or
 *   modify it under the terms of the GNU General Public License
 *   as published by the Free Software Foundation
 *   
 */

#ifndef _H_RVEC
#define _H_RVEC


/* ---------------------------------------------------------------------------- */

#define NRVECS_LOG2			6
#define NUM_RVECS			64		/* = 2 ^ NRVECS_LOG2 */
#define RVEC_MASK			(NUM_RVECS-1)

/* ---------------------------------------------------------------------------- */

#define	RVEC_NOP			0		/* Must be zero */
#define RVEC_ENABLE_FPU			3		/* Load up FPU */

#define RVEC_TRACE_TRAP			6
#define RVEC_ISI_TRAP			7		/* r4=nip, r5=srr1 */
#define RVEC_DSI_TRAP			8		/* r4=dar, r5=srr1 */
#define RVEC_ALIGNMENT_TRAP		9		/* r4=dar, r5=srr1 */
#ifdef EMULATE_603
#define RVEC_DMISS_LOAD_TRAP		10
#define RVEC_DMISS_STORE_TRAP		11
#define RVEC_IMISS_TRAP			12
#endif

#define RVEC_SPR_READ			13 		/* r4=spr#, r5=gprnum */
#define RVEC_SPR_WRITE			14		/* r4=spr#, r5=value */
#define RVEC_PRIV_INST			15		/* r4=opcode */
#define RVEC_ILLEGAL_INST		16		/* r4=opcode */

#define RVEC_UNUSUAL_PROGRAM_EXCEP	17		/* r4=opcode, r5=srr1 */

#define RVEC_ALTIVEC_UNAVAIL_TRAP	18
#define RVEC_ALTIVEC_ASSIST		19		/* r4=srr1 */
#define RVEC_ENABLE_ALTIVEC		20

#define	RVEC_EXIT			21
/* 22 was RVEC_INTERRUPT */
#define RVEC_OSI_SYSCALL		23
#define RVEC_TIMER			24

#define RVEC_IO_READ			25
#define RVEC_IO_WRITE			26

#define RVEC_MSR_POW			27		/* (MSR_POW 0->1) => doze */

/* error/debug */
#define RVEC_UNUSUAL_DSISR_BITS		28		/* dar, dsisr (bit 0,5,9 or 11 was set) */
#define RVEC_MMU_IO_SEG_ACCESS		29		/* IO segment access (more or less unused) */
#define	RVEC_INTERNAL_ERROR		30
#define	RVEC_DEBUGGER			31
#define RVEC_BREAK			32		/* r4 = break_flag */
#define RVEC_BAD_NIP			33		/* r4 = phys_nip */
#define RVEC_OUT_OF_MEMORY		34		/* fatal out of memory... */

#define RVEC_CHECK_IRQS			35		/* check interrupts */


/************************************************************************/
/*	MOL kernel/emulator switch magic				*/
/************************************************************************/

/* magic to be loaded into r4/r5 before the illegal instruction is issued */
#define MOL_ENTRY_R4_MAGIC		0x7ba5
#define MOL_INITIALIZE_FLAG		0x8000
#define MOL_KERNEL_ENTRY_MAGIC		mfmsr	r0	/* any privileged instruction will do */


/************************************************************************/
/*	Kernel definitions						*/
/************************************************************************/

#if defined(__KERNEL__) && !defined( __ASSEMBLY__ )

#define RVEC_RETURN_1( mregsptr, rvec, arg1 ) \
	({ (mregsptr)->rvec_param[0] = (unsigned long)(arg1); \
	  return rvec; })

#define RVEC_RETURN_2( mregsptr, rvec, arg1, arg2 ) \
	({ (mregsptr)->rvec_param[0] = (unsigned long)(arg1); \
	  (mregsptr)->rvec_param[1] = (unsigned long)(arg2); \
	  return rvec; })

#define RVEC_RETURN_3( mregsptr, rvec, arg1, arg2, arg3 ) \
	({ (mregsptr)->rvec_param[0] = (unsigned long)(arg1); \
	  (mregsptr)->rvec_param[1] = (unsigned long)(arg2); \
	  (mregsptr)->rvec_param[2] = (unsigned long)(arg3); \
	  return rvec; })

#endif /* !__ASSEMBLY__ && __KERNEL__ */


/************************************************************************/
/*	userspace definitions						*/
/************************************************************************/

#if !defined(__KERNEL__) || defined(__MPC107__)

#if !defined(__ASSEMBLY__)
#if !defined(__KERNEL__)

typedef struct {
	int		vnum;
	void		*vector;
	const char	*name;
} rvec_entry_t;

extern void 		rvec_init( void );
extern void		rvec_cleanup( void );
extern void		set_rvector( uint vnum, void *vector, const char *vector_name );
extern void		set_rvecs( rvec_entry_t *table, int tablesize );
#endif

/* this struct is private to rvec.c/mainloop.S (offsets are HARDCODED) */
typedef struct {
	int 		(*rvec)( int rvec /*, arguments */ );
	int 		dbg_count;
	const char 	*name;
	int		filler;
} priv_rvec_entry_t;
#endif /* __ASSEMBLY__ */

#define RVEC_ESIZE_LOG2		4	/* 2^n = sizeof(priv_rvec_entry_t) */
#define RVEC_ESIZE		16	/* sizeof(priv_rvec_entry_t) */

#endif /* __KERNEL__ || __MPC107__ */

#endif   /* _H_RVEC */
